lib/internal/process/esm_loader.js

```diff
@@ -31,31 +39,91 @@ exports.importModuleDynamicallyCallback = async function(wrap, specifier) {
   throw new ERR_VM_DYNAMIC_IMPORT_CALLBACK_MISSING();
 };
 
-let ESMLoader = new Loader();
-exports.ESMLoader = ESMLoader;
-
-exports.initializeLoader = initializeLoader;
-async function initializeLoader() {
-  const { getOptionValue } = require('internal/options');
-  const userLoader = getOptionValue('--experimental-loader');
-  if (!userLoader)
-    return;
-  let cwd;
-  try {
-    cwd = process.cwd() + '/';
-  } catch {
-    cwd = 'file:///';
+let processLoader;
+exports.getLoader = () => {
+  if (!processLoader) {
+    processLoader = (async () => {
+      const loader = new Loader();
+
+      const { getOptionValue } = require('internal/options');
+      const userLoaders = getOptionValue('--experimental-loader');
+
+      if (userLoaders.length > 0) {
+        const { emitExperimentalWarning } = require('internal/util');
+        emitExperimentalWarning('--experimental-loader');
+
+        const importedLoaders = [];
+        for (let i = 0; i < userLoaders.length; i += 1) {
+          const ns = await loader.importLoader(userLoaders[i]);
```

Comment:
`ns`?

Reply:
namespace

Reply:
Do you mind renaming it to `namespace` then, or adding a comment? I didn't guess what `ns` meant.